badgecheck/state.py

```diff
@@ -5,5 +5,17 @@
 }
 
 
+# Tasks
 def filter_active_tasks(state):
     return [t for t in state.get('tasks') if not t.get('complete')]
+
+
+# Graph
+def get_node_by_id(state, node_id):
+    """
+    Filter state to return first node that matches the requested id.
+    :param state: state object with "graph" property as a list
+    :param node_id: IRI-format string
```

Comment:
I think this is correct, though I agree with your comment on the variable name for `current_node_id` (which I'm changing to `current_blank_node_number`). An "id" should be an IRI string, and that's how they'll find it in the documents. A caller will never have just the id number, they'll be looking at some other document that will reference an embedded node by an IRI, which might be a HTTP IRI or might be a blank node identifier.